paimon-core/src/main/java/org/apache/paimon/utils/BulkFormatMapping.java

```diff
@@ -35,18 +36,23 @@
 import javax.annotation.Nullable;
 
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.Function;
 
 import static org.apache.paimon.predicate.PredicateBuilder.excludePredicateWithFields;
+import static org.apache.paimon.table.SpecialFields.KEY_FIELD_ID_START;
 
 /** Class with index mapping and bulk format. */
 public class BulkFormatMapping {
 
     @Nullable private final int[] indexMapping;
     @Nullable private final CastFieldGetter[] castMapping;
     @Nullable private final Pair<int[], RowType> partitionPair;
+    @Nullable private final int[] trimmedKeyMapping;
```

Comment:
Now that we have so many mappings, it would be better to add some comments to explain what each mapping means and what columns they store.